media/js/newsletter/form-protocol.es6.js

```diff
@@ -0,0 +1,369 @@
+                    newsletterErrors,
+                    newsletterContent.firstChild
+                );
+            }
+
+            const errorList = document.createElement('ul');
+
+            for (let i = 0; i < errorArray.length; i++) {
+                const item = document.createElement('li');
+                item.appendChild(document.createTextNode(errorArray[i]));
+                errorList.appendChild(item);
+            }
+
+            newsletterErrors.appendChild(errorList);
+            newsletterErrors.style.display = 'block';
+        } else {
+            // no error messages, forward to server for better troubleshooting
+            newsletterForm.setAttribute('data-skip-xhr', true);
+            newsletterForm.submit();
+        }
+    }
+
+    // show sucess message
+    function newsletterThanks() {
+        const thanks = document.getElementById('newsletter-thanks');
+
+        // show thanks message
+        thanks.style.display = 'block';
+    }
+
+    // trigger success event
+    function newsletterSuccess() {
+        if (typeof document.CustomEvent === 'function') {
+            document.dispatchEvent(
+                new CustomEvent('newsletterSuccess', {
+                    bubbles: true,
+                    cancelable: true
+                })
+            );
+        } else if (typeof document.createEvent === 'function') {
+            const event = document.createEvent('Event');
+            event.initEvent('newsletterSuccess', true, true);
+            document.dispatchEvent(event);
+        }
+    }
+
+    // XHR subscribe; handle errors; display thanks message on success.
+    function newsletterSubscribe(evt) {
+        const skipXHR = newsletterForm.getAttribute('data-skip-xhr');
+        if (skipXHR) {
+            return true;
+        }
+        evt.preventDefault();
+        evt.stopPropagation();
+
+        // new submission, clear old errors
+        errorArray = [];
+
+        if (newsletterErrors) {
+            newsletterErrors.style.display = 'none';
+
+            while (newsletterErrors.firstChild) {
+                newsletterErrors.removeChild(newsletterErrors.firstChild);
+            }
+        }
+
+        const fmtHtml = document.getElementById('format-html');
+        const fmtText = document.getElementById('format-text');
+        const fmt = fmtText.checked ? fmtText.value : fmtHtml.value;
+        const email = document.getElementById('id_email').value;
+
+        // Get newsletters by hidden id or checked inputs
+        let newsletters = '';
+        const singleNewsletterForm = document.getElementById('id_newsletters');
+        if (singleNewsletterForm) {
+            newsletters = singleNewsletterForm.value;
+        } else {
+            // MDN suggested Array.from Polyfill: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from#polyfill
+            // Production steps of ECMA-262, Edition 6, 22.1.2.1
+            if (!Array.from) {
```

Comment:
Should we consider adding (the ES5 version) of this polyfill to `lib.js` instead, so all pages / bundles can use the function?

Reply:
I did consider it and came down on individual use as it seemed too much to add for all pages if multi-newsletter was the only use case at the moment. I was on the fence about it though. Happy to move it over if you feel it will be used elsewhere in future.

Reply:
I guess my hesitation to go this route is that this creates a situation where `Array.from()` will thrown an error in IE in some cases still but not others. For example, any script that comes after this will be fine, but scripts before it would not, likewise for non newsletter pages. If we don’t want to include globally yet then I think that’s OK with me, but it’s something we should be mindful of perhaps going forward.

Reply:
> any script that comes after this will be fine, but scripts before it would not, likewise for non newsletter pages

Good point. I think it's worth making global now for consistency.  Re-running integration tests now and will assign you for review once they've passed. Thanks!